samples/build-duration-logger/index.html

```diff
@@ -10,28 +10,63 @@
     const gradleEnterpriseServer = 'https://gradle.my-company.com'; // replace with your Gradle Enterprise base url
     const buildStreamUrl = `${gradleEnterpriseServer}/build-export/v1/builds/since/now?stream`;
     const buildStream = new EventSource(buildStreamUrl);
-    const buildEventStreamUrl = buildId => `${gradleEnterpriseServer}/build-export/v1/build/${buildId}/events?eventTypes=BuildStarted,BuildFinished`;
+
+    class BuildProcessor {
+
+        constructor(gradleEnterpriseServer, maxConcurrentBuildsToProcess) {
+            this.gradleEnterpriseServer = gradleEnterpriseServer;
+            this.buildIds = [];
+            this.numBuildsInProcess = 0;
+            this.maxConcurrentBuildsToProcess = maxConcurrentBuildsToProcess;
+        }
+
+        enqueue(buildId) {
+            this.buildIds.push(buildId);
+            this.processQueuedBuilds();
+        }
+
+        processQueuedBuilds() {
+            if (this.buildIds.length > 0 && this.numBuildsInProcess < this.maxConcurrentBuildsToProcess) {
+                this.processBuild(this.buildIds.shift());
+                setTimeout(this.processQueuedBuilds.bind(this), 0);
+            }
+        }
+
+        buildEventStreamUrl(buildId) {
+            return `${this.gradleEnterpriseServer}/build-export/v1/build/${buildId}/events?eventTypes=BuildStarted,BuildFinished`;
+        }
+
+        processBuild(buildId) {
+            this.numBuildsInProcess++;
+            const buildEventStream = new EventSource(this.buildEventStreamUrl(buildId));
```

Comment:
To meaningfully distinguish `buildStream` from `buildEventStream` we could name the latter `buildDurationStream`.

Reply:
Renamed